lightning/src/ln/channel.rs

```diff
@@ -3121,12 +3121,24 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 	}
 
 	/// Called by channelmanager based on chain blocks being connected.
-	/// Note that we only need to use this to detect funding_signed, anything else is handled by
-	/// the channel_monitor.
+	/// We need to use this to detect funding_signed and outgoing HTLC timed out before we were able
```

Comment:
This commit LGTM. In an ideal world, this commit + its test would be split into a separate PR.

Reply:
Right. I'll remove those two commits from this PR next time I rebase. I originally included them since they carry more meaning with this PR than before (since we may not generate events for the user while waiting on more HTLCs to come in, we need to be absolutely sure that we time them out on time, vs before it was only an issue if the user failed to handle an event), but there's not really any harm in it being separate anyway.

Reply:
Speaking of, I'll add a test for failing-back MPP HTLCs as well as standalone ones when I do so.